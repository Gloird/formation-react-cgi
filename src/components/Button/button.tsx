import React, { Component } from "react";

interface Props {
  texte: string;
  couleurFond?: string;
  couleurTexte?: string;
  couleurBordure?: string;
}
interface State {}

class Button extends Component<Props, State> {
  state = {};

  render() {
    return (
      <button
        style={{
          color: this.props.couleurTexte,
          backgroundColor: this.props.couleurFond,
          border: "1px solid",
          borderColor: this.props.couleurBordure,
        }}
      >
        {this.props.texte}
      </button>
    );
  }
}

export default Button;
