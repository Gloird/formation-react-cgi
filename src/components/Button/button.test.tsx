import { render } from "@testing-library/react";
import Button from "./button";

describe("Button", () => {
  test("Texte est afficher", () => {
    const renderButton = render(<Button texte="Mon texte" />);
    //Bouton est il dans le dom
    expect(renderButton.getByRole("button")).toBeInTheDocument();
    //Bouton contien bien le texte «Mon texte»
    expect(renderButton.getByRole("button")).toHaveTextContent("Mon texte");
    //Bouton contien bien le style «border: "1px solid"»
    expect(renderButton.getByRole("button")).toHaveStyle({
      border: "1px solid",
    });
  });

  test("Couleur du texte", () => {
    const renderButton = render(
      <Button texte="Mon texte" couleurTexte="#3798B8" />
    );
    //Bouton est il dans le dom
    expect(renderButton.getByRole("button")).toBeInTheDocument();
    //Bouton contien bien le texte «Mon texte»
    expect(renderButton.getByRole("button")).toHaveTextContent("Mon texte");
    //Bouton contien bien le style «border: "1px solid",color: "#3798B8"»
    expect(renderButton.getByRole("button")).toHaveStyle({
      border: "1px solid",
      color: "#3798B8",
    });
  });

  test("Couleur de fond", () => {
    const renderButton = render(
      <Button texte="Mon texte" couleurFond="#5CB837" />
    );
    //Bouton est il dans le dom
    expect(renderButton.getByRole("button")).toBeInTheDocument();
    //Bouton contien bien le texte «Mon texte»
    expect(renderButton.getByRole("button")).toHaveTextContent("Mon texte");
    //Bouton contien bien le style «border: "1px solid",backgroundColor: "#5CB837"»
    expect(renderButton.getByRole("button")).toHaveStyle({
      border: "1px solid",
      backgroundColor: "#5CB837",
    });
  });

  test("Couleur de la bordure", () => {
    const renderButton = render(
      <Button texte="Mon texte" couleurBordure="#B8A637" />
    );
    //Bouton est il dans le dom
    expect(renderButton.getByRole("button")).toBeInTheDocument();
    //Bouton contien bien le texte «Mon texte»
    expect(renderButton.getByRole("button")).toHaveTextContent("Mon texte");
    //Bouton contien bien le style «border: "1px solid",backgroundColor: "#5CB837"»
    expect(renderButton.getByRole("button")).toHaveStyle({
      border: "1px solid",
      borderColor: "#B8A637",
    });
  });

  test("Snapshot Button", () => {
    const renderButton = render(<Button texte="Mon texte" />);

    //Permet de s'assurer que l'UI ne change pas.
    expect(renderButton.baseElement).toMatchSnapshot();
  });
});
