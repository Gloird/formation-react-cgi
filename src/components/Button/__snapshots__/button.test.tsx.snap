// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot Button 1`] = `
<body>
  <div>
    <button
      style="border: 1px solid;"
    >
      Mon texte
    </button>
  </div>
</body>
`;
